### DISCUSSION TITLE
How to add if is a leaf node in my nested hierarchies element ?

### DISCUSSION THREAD

**Discussion by NicoSan20** [sequelize/sequelize#14569] [Help & Questions] [answered]
Hi everyone ! Currently I'm using a nested hierarchies like below :

```
const result = await models.Category.findAll({
    where: {
      parent: null
    },
    include: [{
      model: models.Category,
      as: 'children'
    }]
  });
res.send(result);
```

Now, I would like to have on each element found if is an leaf node or not (leaf : true (or false)).

How can I do to get this result ? Any help will be appreciated. Thank you in advance.

**Comment by ephys** [comment, marked as answer]
The following has not been tested

This should return all categories that do not have children (leaves):

```js
const result = await models.Category.findAll({
  where: {
    '$children.id$': null,
  },
  include: [{
    model: models.Category,
    as: 'children',
  }],
});
```

This should return all categories that do have children (not leaves):

```js
const result = await models.Category.findAll({
  include: [{
    model: models.Category,
    as: 'children',
    required: true,
  }],
});
```
